### PR TITLE
MT3-314 #done #comment Add PageAnnouncement component

### PR DESCRIPTION
**Pull request by CJWaszczuk**
Adds a component for displaying announcements on pages.

# What?

When a user is offline and completes a process, they will a receive an announcement that they need to wait to go online to finish their process. This component allows us to do this. 

# Why?

When a user is offline and completes a process, they will a receive an announcement that they need to wait to go online to finish their process. This component allows us to do this. 

# References
`lbh-frontend` [Github](https://github.com/LBHackney-IT/LBH-frontend/tree/master/src/lbh/components/lbh-page-announcement) | [Styleguide](https://lbh-frontend.herokuapp.com/components/lbh-page-announcement)

### CODE REVIEW

**Comment by CJWaszczuk** [comment]
@LBHELewis Could use your input on a few decisions here. Most notably the decisions on what is mandatory and what isn't. I've made title and content required, figured they would be things that would always be in an announcement. Also, whether the docstrings make sense for the implementation.

File: src/components/PageAnnouncement.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * A component for displaying announcements on a page.
+ */
+export const PageAnnouncement: React.FunctionComponent<PageAnnouncementProps> = (
+  props: PageAnnouncementProps
+): JSX.Element => {
+  const { id, className, title, children } = props;
+
+  return (
+    <section id={id} className={classNames("lbh-page-announcement", className)}>
```
Comment (erbridge) [inline src/components/PageAnnouncement.tsx]: No `govuk-page-announcement`?

File: src/components/PageAnnouncement.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+
+import "lbh-frontend/lbh/components/lbh-page-announcement/page-announcement.scss";
```
Comment (erbridge) [inline src/components/PageAnnouncement.tsx]:
```suggestion
import "lbh-frontend/lbh/components/lbh-page-announcement/_page-announcement.scss";
```

I think?

File: src/components/PageAnnouncement.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * The proptypes for {@link PageAnnouncement}.
+ */
+export interface PageAnnouncementProps {
+  id?: string;
+  className?: string;
+  // The title of the announcement
+  title: string;
+  // The content of the announcement
```
Comment (erbridge) [inline src/components/PageAnnouncement.tsx]:
```suggestion
  /**
   * The content of the announcement.
   */
```

File: src/components/PageAnnouncement.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import "lbh-frontend/lbh/components/lbh-page-announcement/page-announcement.scss";
+
+/**
+ * The proptypes for {@link PageAnnouncement}.
+ */
+export interface PageAnnouncementProps {
+  id?: string;
+  className?: string;
+  // The title of the announcement
```
Comment (erbridge) [inline src/components/PageAnnouncement.tsx]:
```suggestion
  /**
   * The title of the announcement.
   */
```

**Review by erbridge — COMMENTED** [review]
Note that docstrings need to be `/** */` comments to be picked up.

File: src/components/PageAnnouncement.tsx (hunk 1 above)
Comment (CJWaszczuk) [reply src/components/PageAnnouncement.tsx]: `lbh-frontend` component only, no govuk implementation

**Review by erbridge — APPROVED** [review]

**Comment by LBHELewis** [comment]
@CJWaszczuk All looks good to me thanks. You made me realise that I also need to add optional class and id to this component in the main pattern library.

**Pull request merged**